packages/react-dev-utils/webpackHotDevClient.js

```diff
@@ -157,14 +157,20 @@ function destroyErrorOverlay() {
   lastOnOverlayDivReady = null;
 }
 
+let rootPathName = url.parse(__webpack_public_path__).pathname || ''; // eslint-disable-line no-undef
+
+if (rootPathName.length > 1) {
+  rootPathName = rootPathName.replace(/\/+$/, '');
+}
+
 // Connect to WebpackDevServer via a socket.
 var connection = new SockJS(
   url.format({
```

Comment:
shouldn't we get `protocol`, `hostname`, `port` from PUBLIC_URL? assume we open diffrent  page than localhost:3000 and only load static files from localhost:3000. [more detail](https://github.com/smmoosavi/djcra#in-frontend-development-mode)